src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -1075,6 +1241,55 @@ private static Vector512<float> FusedMultiplyAdd(Vector512<float> x, Vector512<f
         }
 #endif
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static float HorizontalAggregate<TAggregate>(Vector128<float> x) where TAggregate : struct, IBinaryOperator
+        {
+            float f = x[0];
+            for (int i = 1; i < Vector128<float>.Count; i++)
```

Comment:
Vectorization via hadd is left for future improvements or it's not IEEE754 friendly?

Reply:
hadd handles min/max?

Reply:
It doesn't, but much like `Sum` is a specialized aggregate for addition, there are often optimized ways to handle aggregates for other operations as well.

The general pattern is typically:
```csharp
operation(vector.GetLower(), vector.GetUpper());
```

until you hit `V128`, and then you do:
```csharp
var tmp = Vector128.Shuffle(vector, ...); // swap even/odd indices
operation(vector, tmp);
// repeat up to log2(count) times, so 0 more times for 64-bit T, 1 more for 32-bit T, 2 more for 16-bit T, 3 more for 8-bit T
```

Reply:
Oh sorry I was referring to Sum operation, although, I guess it should be possible to do Max horizontally as well?

Reply:
> I was referring to Sum operation

That's already handled here:
https://github.com/dotnet/runtime/blob/aec06846449c331532022b9c01874c80e5a35fc6/src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs#L1087-L1091